main.cc

```diff
@@ -347,6 +349,60 @@ class sighup_handler {
     }
 };
 
+class sigquit_handler {
+    bool _stopping = false;
+    bool _pending = false;
+    condition_variable _cond;
+    future<> _done;
+
+    sharded<replica::database>& _db;
+
+    future<> execution_loop() {
+        return repeat([this] {
```

Comment:
Why continuation style instead of co-routines?

Reply:
No particular reason, I guess I can use coroutines instead.

Reply:
Switched to coroutines in v3.